--- a/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c
+++ b/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c
@@ -2547,5 +2547,13 @@ static void rk3528_vop2_setup_overlay(struct vop2_video_port *vp)
 		layer_sel &= ~GENMASK(3, 0);
 		layer_sel |= win->data->layer_sel_id[vp->id];
-		if (win->data->phys_id == ROCKCHIP_VOP2_CLUSTER0)
-			port_sel |= vp->id;
+		switch (win->data->phys_id) {
+		case ROCKCHIP_VOP2_CLUSTER0:
+			port_sel |= vp->id & 0x3;
+			break;
+		case ROCKCHIP_VOP2_ESMART0:
+			port_sel |= (vp->id & 0x3) << 16;
+			break;
+		default:
+			break;
+		}
 	}
@@ -2581,14 +2581,13 @@ static const struct vop2_win_data rk3528_vop_win_data[] = {
-		.name = "Cluster0-win0",
-		.phys_id = ROCKCHIP_VOP2_CLUSTER0,
-		.base = 0x1000,
+		.name = "Esmart0-win0",
+		.phys_id = ROCKCHIP_VOP2_ESMART0,
+		.base = 0x0,
 		.possible_vp_mask = BIT(0),
-		.formats = formats_cluster,
-		.nformats = ARRAY_SIZE(formats_cluster),
+		.formats = formats_smart,
+		.nformats = ARRAY_SIZE(formats_smart),
 		.format_modifiers = format_modifiers,
-		.layer_sel_id = { 0, 0xf, 0xf, 0xf },
-		.supported_rotations = DRM_MODE_REFLECT_X | DRM_MODE_REFLECT_Y,
+		.layer_sel_id = { 1, 0xf, 0xf, 0xf },
+		.supported_rotations = DRM_MODE_REFLECT_Y,
 		.type = DRM_PLANE_TYPE_PRIMARY,
-		.feature = WIN_FEATURE_CLUSTER,
-		.max_upscale_factor = 4,
-		.max_downscale_factor = 4,
-		.dly = { 0, 27, 21 },
+		.max_upscale_factor = 8,
+		.max_downscale_factor = 8,
+		.dly = { 27, 45, 48, 35 },
